### ISSUE TITLE
Support scheduling blog posts by setting a publish date in the future

### ISSUE THREAD

**Issue by caendesilva** [hydephp/develop#2001]

**Comment by caendesilva** [comment]
A very easy user-land way to do this is by updating [`blog-post-feed.blade.php`](https://github.com/hydephp/develop/blob/master/packages/framework/resources/views/components/blog-post-feed.blade.php)

```blade
@foreach(MarkdownPost::getLatestPosts() as $post)
    @unless($this->date !== null && $this->date->toDateTime() > now())
        @include('hyde::components.article-excerpt')
    @endunless
@endforeach
```

The posts would still be rendered however, and show up in the sitemap, so that would need to happen on our end. Not sure what is the best way; maybe by ignoring them during discovery? Since that feels very hacky I want to hold off until I get more feedback on if this is something people want.

**Comment by caendesilva** [comment]
Yeah I don't think this works as the post will only get published after a build.

**Comment by emmadesilva** [comment]
Reopening — I think there's a narrower version of this that avoids the original objection.

The core issue with the earlier proposal was that "scheduling" implies temporal automation Hyde can't deliver on: a future date only means something if a build happens to run after it, and most users don't have that infrastructure. That reasoning still holds, and I'm not proposing we change it.

But Hugo and Jekyll both handle the adjacent problem — future-dated posts leaking into the site — without pretending to solve scheduling:

- Both exclude future-dated content **at discovery time** by default (not just in templates), so it never ends up in feeds, sitemaps, or listings.
- Neither promises real-time publishing. Actually "scheduling" a post going live at a specific time is left entirely to external tooling — cron, scheduled GitHub Actions, etc. — rebuilding the site after the date passes.
- Jekyll/Hugo both offer an opt-in flag (`--future` / `--buildFuture`) to include future posts anyway, for people who want them despite the date.

I think we should do the same:

1. Exclude future-dated posts at discovery time (this also resolves the original #2001 complaint that the userland Blade workaround only hides posts from the feed, but they still get rendered and land in the sitemap).
2. Emit a build warning like `X blog posts skipped (future-dated)` so silence isn't mistaken for a bug — this was the actual failure mode we were worried about (user sets a date, nothing happens, assumes Hyde is broken).
3. Document clearly that Hyde is not a server: setting a future date does not schedule anything by itself, and users need their own trigger (cron, CI, manual rebuild) for the post to actually go live on time.

This never claims Hyde delivers "goes live at time X" — it just makes the discovery behavior sane and visible, which is a smaller and more honest feature than what was originally proposed. Users who already know Hyde is an SSG (which is the majority of our audience) should have no trouble understanding they need to trigger their own rebuild.

**Comment by emmadesilva** [comment]
Implemented in v3 https://github.com/hydephp/develop/pull/2571
